certbot/certbot/util.py

```diff
@@ -48,6 +50,200 @@ class CSR(NamedTuple):
     form: str
 
 
+class LooseVersion:
+    """A version with loose rules, i.e. any given string is a valid version number.
+
+    This is based on distutils.version.LooseVersion at
+    https://github.com/python/cpython/blob/v3.10.0/Lib/distutils/version.py#L269
+    but regular comparison is not supported. Instead, the `try_risky_comparison` method is
+    provided, which may return an error if two LooseVersions are 'incomparible'.
+    For example when integer and string version components are present in the same position.
+    """
+
+    def __init__(self, version_string: str) -> None:
+        self.version_components = self.__parse_loose_version(version_string)
+
+    def __parse_loose_version(self, version_string: str) -> List[Union[int, str]]:
+        """Parses a version string into its components.
+
+        This code and the returned tuple is based on the now deprecated
+        distutils.version.LooseVersion class from the Python standard library.
+        Two LooseVersion classes and two lists as returned by this function
+        should compare in the same way. See
+        https://github.com/python/cpython/blob/v3.10.0/Lib/distutils/version.py#L205-L347.
+
+        :param str version_string: version string
+
+        :returns: list of parsed version string components
+        :rtype: list
+        """
+        components: List[Union[int, str]]
+        components = [x for x in _VERSION_COMPONENT_RE.split(version_string)
+                              if x and x != '.']
+        for i, obj in enumerate(components):
+            try:
+                components[i] = int(obj)
+            except ValueError:
+                pass
+        return components
+
+    class Comparison(Enum):
```

Comment:
I think the rest of this class can probably be simplified. If the type annotation on the comparison function is also `LooseVersion`, we arguably don't have to check for any other type. After doing that, I think the rest of the comparison function could be as simple as:
```
return (self.version_components > other.version_components) - (self.version_components < other.version_components)
```
See https://docs.python.org/3.0/whatsnew/3.0.html#ordering-comparisons which should probably be referenced in a code comment since that's pretty magical.

Maybe we'd want to catch and reraise the `TypeError` with a different error message, but that otherwise seems like it should be sufficient to me. What do you think?

Reply:
This would simplify the implementation considerably, but this would give a slightly different result than the current implementation. "2.0" and "2.0.0" would currently be equal, but the simplified version "2.0.0" > "2.0". If this doesn't matter I think we should go with the simplified version.

Reply:
Sorry for the late response, I have been quite busy lately. 
Your proposed approach passes all my tests :+1: 
I'll implement it with a `ValueError` reraise.